# 🧠 Kor-DEEPression
### 📊 Step 2-2. EDA 및 시각화 분석
> 🍷 작성자 : Codestates AIB 15기 정경재 (Kyung Jae, Cheong)
---

### 💻 라이브러리 및 데이터 불러오기

- Library Import (라이브러리 불러오기)

In [3]:
# Library Import
import numpy as np
import pandas as pd

import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as plex
import plotly.graph_objects as go

# Matplotlib setting for VSCode
plt.rcParams['font.family'] = 'Malgun Gothic'
plt.rcParams['axes.unicode_minus'] = False
# Renderer setting for VSCode & Github
import plotly.io as pio
pio.renderers.default = 'vscode+png'

- EDA Data Import (EDA 데이터 불러오기)

In [8]:
# csv파일을 DataFrame으로 불러오기
df = pd.read_csv('downloads/EDA.csv')

### 🔎 데이터 확인

- Shape, Head, Tail (모양 확인)

In [9]:
# Shape 확인
print(f"Data Shape : {df.shape}")
# Head와 Tail data 확인
display(df.head())
display(df.tail())

Data Shape : (16570, 23)


id  year  age        BMI     sex education household   marital  \
0  A209716213  2014   30  28.939858  Female     대졸 이상       2세대        미혼   
1  A209716216  2014   29  28.659708    Male     대졸 이상       2세대        미혼   
2  A209724716  2014   65  20.188238    Male     초졸 이하       1세대  기혼(유배우자)   
3  A209731516  2014   66  24.145909    Male     대졸 이상       1세대  기혼(유배우자)   
4  A209736615  2014   41  30.598080  Female        고졸       2세대  기혼(유배우자)   

       economy subj_health  ...  high_bp  diabetes  dyslipidemia  \
0  실업자/비경제활동인구          보통  ...        0         0             0   
1  실업자/비경제활동인구          보통  ...        0         0             0   
2          취업자          보통  ...        1         1             0   
3  실업자/비경제활동인구          좋음  ...        0         0             0   
4  실업자/비경제활동인구          보통  ...        0         0             0   

       drk_freq  drk_amount     smoke     stress depression MDD Depr_MDD  
0       월 1회 미만        1-2잔  비흡연/과거흡연  적게 느끼는 편임          0   0       정상  
1     월 2-4회 정도        5-6잔  비흡연/과거흡연  거의 느끼지 않음          0   0       정상  
2   평생 마셔본 적 없음      전혀 안마심      현재흡연  적게 느끼는 편임          0   0       정상  
3  1년간 마셔본 적 없음      전혀 안마심  비흡연/과거흡연  적게 느끼는 편임          0   0       정상  
4       주 4회 이상      10잔 이상  비흡연/과거흡연  대단히 많이 느낌          1   0    경도우울증  

[5 rows x 23 columns]

id  year  age        BMI     sex education household   marital  \
16565  R804363202  2020   26  17.570804  Female     대졸 이상       2세대        미혼   
16566  R804366601  2020   61  23.913275  Female        고졸       2세대  기혼(유배우자)   
16567  R804368302  2020   24  26.129927  Female     대졸 이상       2세대        미혼   
16568  R804371701  2020   65  25.941572    Male     대졸 이상       1세대  기혼(유배우자)   
16569  R804371702  2020   57  22.287517  Female        고졸       1세대  기혼(유배우자)   

           economy subj_health  ...  high_bp  diabetes  dyslipidemia  \
16565          취업자          보통  ...        0         0             0   
16566          취업자          보통  ...        0         0             0   
16567          취업자          보통  ...        0         0             0   
16568  실업자/비경제활동인구          좋음  ...        0         0             1   
16569          취업자          나쁨  ...        0         0             0   

        drk_freq  drk_amount     smoke     stress depression MDD Depr_MDD  
16565    월 1회 정도        3-4잔  비흡연/과거흡연  많이 느끼는 편임          0   0       정상  
16566    월 1회 미만        1-2잔  비흡연/과거흡연  적게 느끼는 편임          0   0       정상  
16567    주 4회 이상      10잔 이상      현재흡연  적게 느끼는 편임          0   0       정상  
16568  월 2-4회 정도        5-6잔  비흡연/과거흡연  적게 느끼는 편임          0   0       정상  
16569    월 1회 미만        1-2잔  비흡연/과거흡연  적게 느끼는 편임          0   0       정상  

[5 rows x 23 columns]

- Check Null values & Duplicate values (결측값 및 중복값)

In [13]:
# 결측치 확인
print(f"Sum of Null Values : {df.isnull().sum().sum()}")
# 중복값 확인 (고유식별코드인 "id" column만 확인)
print(f"Sum of Duplicated Values : {df.id.duplicated().sum()}")

Sum of Null Values : 0
Sum of Duplicated Values : 0


- column's dtype (column 타입)

In [14]:
# Column 별 dtype을 확인
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16570 entries, 0 to 16569
Data columns (total 23 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   id            16570 non-null  object 
 1   year          16570 non-null  int64  
 2   age           16570 non-null  int64  
 3   BMI           16570 non-null  float64
 4   sex           16570 non-null  object 
 5   education     16570 non-null  object 
 6   household     16570 non-null  object 
 7   marital       16570 non-null  object 
 8   economy       16570 non-null  object 
 9   subj_health   16570 non-null  object 
 10  limitation    16570 non-null  int64  
 11  modality      16570 non-null  int64  
 12  w_change      16570 non-null  int64  
 13  high_bp       16570 non-null  int64  
 14  diabetes      16570 non-null  int64  
 15  dyslipidemia  16570 non-null  int64  
 16  drk_freq      16570 non-null  object 
 17  drk_amount    16570 non-null  object 
 18  smoke         16570 non-nu

- Variable Classification (변수 분류)

In [49]:
# 식별코드 "id"를 제외한 변수들의 수치형 변수 및 범주형 변수 갯수 확인
# describe() 함수의 include 기능을 활용하여 갯수를 산출함(분석가능 범주를 확인)
print(f"Sum of Numerical Data : {df.iloc[:,1:].describe(include=['number']).shape[1]}")
print(f"Sum of Categorical Data : {df.iloc[:,1:].describe(include=['object']).shape[1]}")

Sum of Numerical Data : 11
Sum of Categorical Data : 11


In [50]:
# 수치형 변수 기술통계(descriptive statistics)
df.iloc[:,1:].describe(include=['number']).round(1)

year      age      BMI  limitation  modality  w_change  high_bp  \
count  16570.0  16570.0  16570.0     16570.0   16570.0   16570.0  16570.0   
mean    2017.2     51.4     24.0         0.1       0.2       0.1      0.3   
std        2.2     16.7      3.6         0.3       0.4       0.3      0.5   
min     2014.0     19.0     14.1         0.0       0.0       0.0      0.0   
25%     2016.0     38.0     21.5         0.0       0.0       0.0      0.0   
50%     2018.0     52.0     23.7         0.0       0.0       0.0      0.0   
75%     2020.0     65.0     26.0         0.0       0.0       0.0      1.0   
max     2020.0     80.0     62.6         1.0       1.0       1.0      1.0   

       diabetes  dyslipidemia  depression      MDD  
count   16570.0       16570.0     16570.0  16570.0  
mean        0.1           0.3         0.2      0.1  
std         0.3           0.5         0.4      0.2  
min         0.0           0.0         0.0      0.0  
25%         0.0           0.0         0.0      0.0  
50%         0.0           0.0         0.0      0.0  
75%         0.0           1.0         0.0      0.0  
max         1.0           1.0         1.0      1.0

In [51]:
# 수치형 변수의 분류를 위한 unique값 갯수 확인
numeric_columns = df.iloc[:,1:].describe(include=['number']).columns
df[numeric_columns].nunique()

year                4
age                62
BMI             15328
limitation          2
modality            2
w_change            2
high_bp             2
diabetes            2
dyslipidemia        2
depression          2
MDD                 2
dtype: int64

In [53]:
num_continuous = ((df[numeric_columns].nunique()<5)==False).sum()
# 연속형 변수의 갯수
print(f"Sum of Continous Data : {num_continuous}")
# 범주형 변수의 갯수(수치형 데이터 포함)
print(f"Sum of Categorical Data : {df.iloc[:,1:].shape[1] - num_continuous}")

Sum of Continous Data : 2
Sum of Categorical Data : 20


##### 💡 데이터 확인 결과 요약
- Data는 16570개의 row와 23개의 column으로 이루어져 있음.

- 전처리를 거친 데이터이기 때문에 결측데이터 및 중복데이터는 존재하지 않았음.

- 식별코드인 id를 제외하고 수치형 및 범주형 데이터 갯수는 다음과 같음.

  - 수치형 변수(Numerical Data) : 총 11개
  
  - 범주형 변수(Categorical Data) : 총 11개

- 수치형 변수의 unique값의 개수를 고려하여 재분류 한 결과는 다음과 같음.

  - 연속형 변수(Continous Data) : 총 2개 ('age', 'BMI')
  
  - 범주형 변수(Categorical Data) : 총 20개